libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthKeychainCredentials.m

```diff
@@ -80,65 +113,33 @@ - (NSData *)tokenForService:(NSString *)service
-        return [[NSString alloc] initWithData:decryptedData encoding:NSUTF8StringEncoding];
-    } else {
-        return [[NSString alloc] initWithData:accessTokenData encoding:NSUTF8StringEncoding];
-    }
-}
-
-- (void)setAccessToken:(NSString *)token withEncryptionKey:(NSData *)encryptionKey {
-    NSData *tokenData = ([token length] > 0 ? [token dataUsingEncoding:NSUTF8StringEncoding] : nil);
-    if (tokenData != nil) {
-        if (self.isEncrypted) {
-            tokenData = [SFSDKEncryptor encryptData:tokenData key:encryptionKey error:nil];
-        }
-    }
-    
-    BOOL updateSucceeded = [self updateKeychainWithTokenData:tokenData forService:kSFOAuthServiceAccess];
-    if (!updateSucceeded) {
-        [SFSDKCoreLogger w:[self class] format:@"%@:%@ - Failed to update access token.", [self class], NSStringFromSelector(_cmd)];
-    }
-}
-
-- (NSString *)refreshTokenWithEncryptionKey:(NSData *)encryptionKey {
-    NSData *refreshTokenData = [self tokenForService:kSFOAuthServiceRefresh];
-    if (!refreshTokenData) {
+- (NSString *)decryptedTokenForService:(NSString *)service {
+    NSData* encryptionKey = [self encryptionKeyForService:service];
+    NSData *data = [self tokenForService:service];
+    if (!data) {
         return nil;
     }
     
     if (self.isEncrypted) {
-        NSData *decryptedData = [SFSDKEncryptor decryptData:refreshTokenData key:encryptionKey error:nil];
+        NSData *decryptedData = [SFSDKEncryptor decryptData:data key:encryptionKey error:nil];
         return [[NSString alloc] initWithData:decryptedData encoding:NSUTF8StringEncoding];
     } else {
-        return [[NSString alloc] initWithData:refreshTokenData encoding:NSUTF8StringEncoding];
+        return [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
     }
 }
 
-- (void)setRefreshToken:(NSString *)token withEncryptionKey:(NSData *)encryptionKey {
+- (void)encryptToken:(NSString *)token forService:(NSString *)service {
+    NSData* encryptionKey = [self encryptionKeyForService:service];
     NSData *tokenData = ([token length] > 0 ? [token dataUsingEncoding:NSUTF8StringEncoding] : nil);
     if (tokenData != nil) {
         if (self.isEncrypted) {
             tokenData = [SFSDKEncryptor encryptData:tokenData key:encryptionKey error:nil];
         }
-    } else {
-        self.instanceUrl = nil;
-        self.communityId  = nil;
-        self.communityUrl = nil;
-        self.issuedAt    = nil;
-        self.identityUrl = nil;
     }
     
-    BOOL updateSucceeded = [self updateKeychainWithTokenData:tokenData forService:kSFOAuthServiceRefresh];
+    BOOL updateSucceeded = [self updateKeychainWithTokenData:tokenData forService:service];
     if (!updateSucceeded) {
-        [SFSDKCoreLogger w:[self class] format:@"%@:%@ - Failed to update refresh token.", [self class], NSStringFromSelector(_cmd)];
+        [SFSDKCoreLogger w:[self class] format:@"%@:%@ - Failed to update %@.", [self class], service];
```

Comment:
the format string looks like it has more string specifiers than string arguments, maybe this instead?
`[SFSDKCoreLogger w:[self class] format:@"%@:%@ - Failed to update %@.", [self class], NSStringFromSelector(_cmd), service];`